yt_dlp/extractor/googledrive.py

```diff
@@ -289,55 +313,91 @@ def add_source_format(urlh):
-content-transfer-encoding: binary
-
-GET %s
 
---{_BOUNDARY}
-'''
+    def _extract_json_meta(self, webpage, video_id, dsval=None, hashval=None, name=None, **kwargs):
+        """
+        Uses regex to search for json metadata with 'ds' value(0-5) or 'hash' value(1-6)
+        from the webpage.
+            logged out  folder info:ds0hash1;          items:ds4hash6
+            logged in   folder info:ds0hash1;          items:ds5hash6
+            my-drive    folder info:ds0hash1/ds0hash4; items:ds5hash6
+        For example, if the webpage contains the line below, the empty data array
+        can be got by passing dsval=3 or hashval=2 to this method.
+            AF_initDataCallback({key: 'ds:3', hash: '2', data:[], sideChannel: {}});
+        """
+        _ARRAY_RE = r'\[(?s:.+)\]'
+        _META_END_RE = r', sideChannel: \{\}\}\);'  # greedy match to deal with the 2nd test case
+        if dsval is not None:
+            if not name:
+                name = f'webpage JSON metadata ds:{dsval}'
+            return self._search_json(
+                rf'''key\s*?:\s*?(['"])ds:\s*?{dsval}\1,[^\[]*?data:''', webpage, name, video_id,
+                end_pattern=_META_END_RE, contains_pattern=_ARRAY_RE, **kwargs)
+        elif hashval is not None:
+            if not name:
+                name = f'webpage JSON metadata hash:{hashval}'
+            return self._search_json(
+                rf'''hash\s*?:\s*?(['"]){hashval}\1,[^\[]*?data:''', webpage, name, video_id,
+                end_pattern=_META_END_RE, contains_pattern=_ARRAY_RE, **kwargs)
 
-    def _call_api(self, folder_id, key, data, **kwargs):
-        response = self._download_webpage(
-            'https://clients6.google.com/batch/drive/v2beta',
-            folder_id, data=data.encode(),
-            headers={
-                'Content-Type': 'text/plain;charset=UTF-8;',
-                'Origin': 'https://drive.google.com',
-            }, query={
-                '$ct': f'multipart/mixed; boundary="{self._BOUNDARY}"',
-                'key': key,
-            }, **kwargs)
-        return self._search_json('', response, 'api response', folder_id, **kwargs) or {}
+    def _real_extract(self, url):
+        def item_url_getter(item, video_id):
+            if not isinstance(item, list):
+                return None
+            available_IEs = (GoogleDriveFolderIE, GoogleDriveIE)  # subfolder or item
+            if 'application/vnd.google-apps.shortcut' in item:  # extract real link
+                entry_url = traverse_obj(
+                    item,
+                    (..., ..., lambda _, v: any(ie.suitable(v) for ie in available_IEs), any))
+            else:
+                entry_url = traverse_obj(
+                    item,
+                    (lambda _, v: any(ie.suitable(v) for ie in available_IEs), any))
+            if not entry_url:
+                return None
+            return self.url_result(entry_url, video_id=video_id, video_title=item[2])
```

Comment:
```suggestion
            return self.url_result(entry_url, video_id=video_id, video_title=traverse_obj(item, 2))
```
this should be made non-fatal